prov/util/src/util_buf.c

```diff
@@ -67,28 +77,47 @@ int util_buf_grow(struct util_buf_pool *pool)
 	ret = ofi_memalign((void **)&buf_region->mem_region, pool->attr.alignment,
 			     pool->attr.chunk_cnt * pool->entry_sz);
 	if (ret)
-		goto err;
+		goto err1;
 
 	if (pool->attr.alloc_hndlr) {
 		ret = pool->attr.alloc_hndlr(pool->attr.ctx,
 					     buf_region->mem_region,
 					     pool->attr.chunk_cnt * pool->entry_sz,
 					     &buf_region->context);
 		if (ret)
-			goto err;
+			goto err2;
+	}
+
+	if (pool->attr.track_index &&
+	    (!pool->regions_table ||
+	     pool->regions_cnt >= UTIL_BUF_POOL_REGION_CHUNK_CNT)) {
+		pool->regions_table = realloc(pool->regions_table,
```

Comment:
We need to do something like this:
```
new_table = realloc(...);
if (!new_table)
    goto err3;
pool->regions_table = new_table;
```
or we lose the regions_table if realloc() fails.